Add TokenExpiration to Response

w-haibara opened the pull request:
GitHub has been updated to allow you to set the expiration date of your personal access token. And the expiration date is now displayed in the `GitHub-Authentication-Token-Expiration` header of the GItHub API. 
https://github.blog/changelog/2021-07-26-expiration-options-for-personal-access-tokens/

This PR add `TokenExpiration` field to Response.

gmlewis commented on example/tokenauth/main.go:
```diff
@@ -39,7 +40,12 @@ func main() {
+	
+	// If a Token Expiration has been set, it will be displayed.
+	if !resp.TokenExpiration.IsZero() {
+		log.Printf("Token Expiration: %v\n", resp.TokenExpiration.String())
```
```suggestion
		log.Printf("Token Expiration: %v\n", resp.TokenExpiration)
```

gmlewis commented on github/github.go:
```diff
@@ -551,6 +557,16 @@ func parseRate(r *http.Response) Rate {
+func parseTokenExpiration(r *http.Response) Timestamp {
+	var exp Timestamp
+	if v := r.Header.Get(headerTokenExpiration); v != "" {
+		t, _ := time.Parse("2006-01-02 03:04:05 MST", v)
```
```suggestion
		if t, err := time.Parse("2006-01-02 03:04:05 MST", v); err == nil {
			exp = Timestamp{t.Local()}
		}
```

gmlewis reviewed (changes requested):
Thank you, @w-haibara !

In addition to the comments below, please add a new unit test for the new `parseTokenExpiration` function.

Also please make sure to run `gofmt` on the edited files so that their formatting is standardized.

See CONTRIBUTING.md in the top-level of this repo for more information.

w-haibara commented:
Hi, @gmlewis ! Thanks for the thoughtful review.
I've fixed the areas you commented on and added a test. I apologize again for the unformatted commits. I've fixed that as well.

gmlewis commented on github/github_test.go:
```diff
@@ -2206,3 +2206,36 @@ func TestRateLimits_Marshal(t *testing.T) {
+				Request: &http.Request{},
+				Header:  http.Header{},
+			},
+			header: "2021-09-03 02:34:04 UTC",
```
Can you please add one more test with a completely garbage header and show how it behaves in that case?

gmlewis commented on github/github_test.go:
```diff
@@ -2206,3 +2206,36 @@ func TestRateLimits_Marshal(t *testing.T) {
+			want:   Timestamp{},
+		},
+		{
+			res: &http.Response{
```
Since `res` starts out the same in every test case, you can remove it from this struct and initialize a new fresh one below within the test loop.

gmlewis reviewed (changes requested):
Thank you, @w-haibara !

gmlewis reviewed (approved):
Thank you, @w-haibara !
LGTM.

Awaiting second LGTM before merging.

w-haibara commented:
Thank you, @gmlewis.
This is almost the first OSS contributor for me. Your review has helped me a lot!

gmlewis commented:
> Thank you, @gmlewis.
> This is almost the first OSS contributor for me. Your review has helped me a lot!

Thank you, for your contributions, @w-haibara !  Your help is greatly appreciated!

Parker77 reviewed (approved):
LGTM!
